dbarashinvd/sonic-buildimage
# fix code for warm reboot to work with FW controlled ports

#### Why I did it
fix the code to work also after warm reboot
to work with FW controlled ports
in warm reboot the control state of each port does not change unlike reboot or fast boot

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
check procfs cmdline if warm reboot done
this is due to the fact pmon dont recognize warm reboot when it's taking place since pmon is loaded after warm reboot is finished
if warm reboot done - check in static detection part for each port if it's FW controlled. if so - leave it this way and stop the state machine flow (set it to final state).

#### How to verify it
boot a switch with CMIS host management with at least one FW controlled port (non active cables or non cmis cables)
then run warm reboot
verify no errors of sysfs reading appears for control sysfs



#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
